students_proj/src/main/java/com/musala/simple/students/Main.java

```diff
@@ -0,0 +1,279 @@
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Reads the backup Json file (if exists) and prints the
+	 * students information from it.
+	 *
+	 */
+	private static void readBackupJson() throws ResourceDoesNotExistException {
+		File backupFile = new File(BACKUP_JSON_PATH);
+
+		if (!backupFile.exists()) {
+			throw new ResourceDoesNotExistException(ERROR_NO_BACKUP_JSON);
+		}
+		String backupStudentsJson = readFile(BACKUP_JSON_PATH);
+
+		StudentWrapper studentWrapper = new Gson().fromJson(backupStudentsJson, StudentWrapper.class);
+		printStudents(Arrays.asList(studentWrapper.students));
+	}
+
+	/**
+	 * Updates the backup file by adding the new students. If the backup
+	 * file does not exist the {@link Main#createBackupJson(String)} is called
+	 * and the backup is created with the new students.
+	 *
+	 * @param studentsJsonInfo A Json String, containing the students
+	 * to be added to the backup file.
+	 */
+	public static void updateBackupFile(String studentsJsonInfo) {
+		File backupFile = new File(BACKUP_JSON_PATH);
+
+		if (!backupFile.exists()) {
+			System.out.println(INFO_CREATING_BACKUP);
+			createBackupJson(studentsJsonInfo);
+		} else {
+			String backupFileInfo = readFile(BACKUP_JSON_PATH);
+			Student[] backupStudents = new Gson().fromJson(backupFileInfo, StudentWrapper.class).students;
+			Student[] newStudents = new Gson().fromJson(studentsJsonInfo, StudentWrapper.class).students;
+			
+			StudentGroup sGroup = new StudentGroup();
+			sGroup.fillStudentGroup(backupStudents);
+			sGroup.fillStudentGroup(newStudents);
+		
+			StudentWrapper allStudentsWrapper = new StudentWrapper();
+			int sGroupSize = sGroup.getStudents().values().size();
+			allStudentsWrapper.students = sGroup.getStudents().values().toArray(new Student[sGroupSize]);
+			createBackupJson(new Gson().toJson(allStudentsWrapper));
+		}
+	}
+
+	/**
+	 * Tries to parse a String using the
+	 * {@link Gson#fromJson(com.google.gson.JsonElement, Class) method, which throws
+	 * an Exception if the parameter String is not a valid Json.
+	 *
+	 * @param Json A String that the user wants to validate as JSON
+	 * @return validation of weather the given String is a valid JSON
+	 */
+	private static boolean isValidJson(String Json) {
+		if (Json.equals(null)) {
+			return false;
+		}
+		try {
+			new Gson().fromJson(Json, Object.class);
+			return true;
+		} catch (JsonSyntaxException ex) {
+			return false;
+		}
+	}
+
+	/**
+	 * TryParses the second argument of the user input (command line). If the
+	 * argument is a valid integer the method returns it. Else the method notifies
+	 * the user that the argument is not a number and returns -1 (Invalid user id).
+	 * If the args array does not contain a second element the method also returns
+	 * -1 (Invalid user id).
+	 *
+	 * @param args the command line arguments provided by the user
+	 * @return an int value specifying student's id
+	 */
+	private static int readStudentId(String[] args) {
+		int studentId = -1;
```

Comment:
Although some API's rely on the logic returning -1 as an integer denoting some wrong behavior, I would suggest to avoid this logic, because it requires that the caller handles the case of -1 being returned.

A better solution which I propose is to define a new package com.musala.simple.students.exception,  create your own exception e.g. `InvalidStudentIdException` and redefine the readStudentId method as follows `private static int readStudentId(String[] args) throws InvalidStudentIdException`. The method logic, on the other hand, will be changed to throw the exception in the case it currently returns -1. In this way, the caller can simply do a try-catch and not do any overhead checks for -1. ;)

Reply:
I started implementing your suggestion, but then I realized I don't want to throw an exception if no second argument was given (the user is not obligated to request user details by id). I could check for a second argument before calling the readStudentId() method, but I decided to remove the whole method instead and implemented a different approach. However, I did create my own exception as you proposed.